src/main/resources/jenkins/plugins/openstack/compute/OpenStack.groovy

```diff
@@ -0,0 +1,89 @@
+package jenkins.plugins.openstack.compute
+
+import jenkins.plugins.openstack.compute.internal.DestroyMachine
+import jenkins.plugins.openstack.compute.internal.Openstack
+import org.openstack4j.model.compute.Server
+
+class OpenStack implements Serializable {
+
+    private org.jenkinsci.plugins.workflow.cps.CpsScript script
+
+    public OpenStack(org.jenkinsci.plugins.workflow.cps.CpsScript script) {
+            this.script = script
+    }
+
+    public Node server(String cloudName, String template) {
+        new Node(this, cloudName, template)
+    }
+
+    public static class Node implements Serializable {
+
+        private final OpenStack openstack
+        public final String cloudName
+        public final String template
+        private Server server = null
+
+        private Node(OpenStack openstack, String cloudName, String template) {
+            this.openstack = openstack
+            this.cloudName = cloudName
+            this.template = template
+        }
+
+        public void destroy() {
+            if (server != null) {
+                DestroyMachine dm
+                dm = new DestroyMachine(cloudName, server.getId())
+                dm.dispose()
+                server = null
+            } else {
+                //Cant destroy what isn't created
+            }
+        }
+
+        public String boot() throws IllegalArgumentException {
+            boot("unlimited")
```

Comment:
I am -1 on defaulting to unlimited - we want users to be careful about this. This overload should scope to current build or whatever else it makes sense in pipeline. Do you perform teardown at some later point in the same pipeline or externally?